apps/nextjs/src/app/aila/[id]/share/page.tsx

```diff
@@ -25,8 +26,10 @@ export async function generateMetadata({
 }
 
 function userCanShare(user: User) {
-  const isDemoUser = Boolean(user.publicMetadata.isDemoUser ?? "true");
-
+  if (!demoUsers.isDemoStatusSet(user)) {
+    return false;
+  }
+  const isDemoUser = Boolean(user.publicMetadata.labs.isDemoUser ?? "true");
```

Comment:
does this need to be publicMetadata.labs?.isDemoUser or is this safe at this point?

Reply:
The type of `isDemoStatusSet` is

```ts
  isDemoStatusSet(user: LabsUser): user is UserWithDemoStatus
 ```

So that's already asserted here!

Reply:
I guess my concern is that we might be trusting what is coming through here, and it might be correct in the types, but I'm not sure we validate the actual data from Clerk?

Reply:
Do you mean that isDemoStatusSet could be accidentally sanitising incorrect data with a correct type?


Here's the definition:
```ts
  isDemoStatusSet(user: LabsUser): user is UserWithDemoStatus {
    const labsMetadata = user.publicMetadata.labs || {};
    return "isDemoUser" in labsMetadata && labsMetadata.isDemoUser !== null;
  }
```